### DISCUSSION TITLE
The number of messages in unacknowledged state for a subscription

### DISCUSSION THREAD

**Discussion by grimmjowK** [apache/pulsar#23139] [Q&A] [answered]
### background
right now, I see many metrics about unacknowledged messages or consume delay, but I haven't got a metric about 'The number of messages in unacknowledged state for a subscription'
### metrics
* pulsar_subscription_back_log\
it's a entry level metric 
* pulsar_storage_backlog_size\
it's a topic level, not subscription level
### question
how can I get the number of messages in unacknowledged state for a subscription ?

**Comment by lhotari** [comment, marked as answer]
You can use the Pulsar Admin API's stats endpoint (In the pulsar-admin command line too `pulsar-admin topics stats`) to get the information. More details in https://pulsar.apache.org/docs/3.3.x/administration-stats/ .

> unackedMessages | The number of unacknowledged messages for the subscription, where an unacknowledged message is one that has been sent to a consumer but not yet acknowledged. This field is only meaningful when using a subscription that tracks individual message acknowledgment.
> -- | --
>

**Reply by grimmjowK** [reply]
thank U for your reply
it's helpful to me, however, I was told that this stat may cause greater pressure on pulsar ？

**Reply by lhotari** [reply]
> thank U for your reply
> it's helpful to me, however, I was told that this stat may cause greater pressure on pulsar ？

It depends on many factors. You could test it yourself and see how your use of the API performs.

**Reply by grimmjowK** [reply]
Thank U very much
